imports/client/components/PuzzlePage.tsx

```diff
@@ -738,6 +1046,12 @@ const PuzzlePageMetadata = ({
     </PuzzleMetadataExternalLink>
   ) : null;
 
+  const imageInsert = isDesktop && document && document.provider === 'google' && document.value.type === 'spreadsheet' && (
+    <PuzzleMetadataRow>
```

Comment:
Have we tried fitting this into the design in other rows before putting it on a row of its own?  I'm not like set against putting it on a row by itself, but I want to be real careful about spending another whole row of vertical space like this on every page for what appears to be a single button.

Reply:
There's plenty of space in the first row (especially since we know we're dealing with a desktop layout). It'd look something like this:

<img width="1441" alt="Screenshot 2022-12-31 at 9 02 47 PM" src="https://user-images.githubusercontent.com/28167/210161455-fc850862-ffad-41c3-871e-f3ff3af8d95f.png">

That was how I originally prototyped it, actually, but it felt pretty distant from the spreadsheet iframe it was operating on.

Reply:
Hmmm, fair.  What if you put it in the row with the tags, before the tags?  Then it's still sheet-adjacent, but doesn't consume a whole row by itself?  (After the tags might lead to some confusion with the plus/minus buttons right next to the insert image button.)

Reply:
Hmm. Yeah, that doesn't seem terrible:

<img width="1391" alt="Screenshot 2023-01-01 at 9 51 05 AM" src="https://user-images.githubusercontent.com/28167/210180221-5f76f5d4-0735-48be-9682-4acf05f7dc5f.png">

Reply:
I'm a little concerned that's going to look strange when tags wrap to a second row. Would you consider pulling it to the right, flowed separately from the taglist (and I guess bottom aligned)? It's probably going to be a little awkward wherever it ends up, since really it belongs below.

Separately, I think this should be styled as a button, as it is an action.

This is an amazing feature by the way. Thank you.

Reply:
I think it's definitely true that it'll be a little awkward anywhere. I thought it looked OK when the tag list started to wrap?

<img width="847" alt="Screenshot 2023-01-01 at 10 15 32 AM" src="https://user-images.githubusercontent.com/28167/210180824-92b6d719-3ddb-44e4-acdc-aade4c5db322.png">

Here's what it looks like if I pull it to the right and make it a button:

<img width="1088" alt="Screenshot 2023-01-01 at 10 22 18 AM" src="https://user-images.githubusercontent.com/28167/210181028-cf347277-39ca-479a-8cfc-6afd04228ad9.png">

<img width="885" alt="Screenshot 2023-01-01 at 10 22 23 AM" src="https://user-images.githubusercontent.com/28167/210181032-4e64deb1-287b-4736-bc55-da3147ba2658.png">
<img width="806" alt="Screenshot 2023-01-01 at 10 22 26 AM" src="https://user-images.githubusercontent.com/28167/210181036-60d4748b-2bc8-411f-8b6c-404da971a5d1.png">

Reply:
For the record, I think basically all of the prototypes at this point have been Fine™ and none of them have been amazing, so I don't really have a strong preference among any of them.

Reply:
Yeah, anything that breaks up the tag row feels really awkward… Thanks for trying it.

My vote is just to put it with the Guess and Edit buttons, and accept that it’s a little more separated from the sheet itself.

Reply:
Works for me:

<img width="1114" alt="Screenshot 2023-01-01 at 9 09 18 PM" src="https://user-images.githubusercontent.com/28167/210196730-7cb04577-c51c-404a-8b5a-1eb3f8d70269.png">

(Note that the "edit" button will only show up for operators; non-operators will only see the insert image/guess buttons.)